### PR TITLE
crd: Add MultiClusterService & MultiClusterIngress CRDs and client code

### PR DESCRIPTION
**Pull request by erickhernandezascencio**
## Checklist
- [X] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

Codegen for GlobalService and MultiClusterIngress

### CODE REVIEW

File: pkg/apis/azuremulticlusteringress/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AzureMultiClusterIngress is the resource AGIC is watching in MultiCluster mode
+type AzureMultiClusterIngress struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec v1.IngressSpec `json:"spec"`
```
Comment (akshaysngupta) [inline pkg/apis/azuremulticlusteringress/v1alpha1/types.go:23]: If MCI Spec is Ingress, we won't be able to extend the spec

File: crds/AzureMultiClusterIngress.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.2.5
+  creationTimestamp: null
+  name: azuremulticlusteringresses.networking.aks.io
```
Comment (akshaysngupta) [inline crds/AzureMultiClusterIngress.yaml:8]: maybe remove `azure` from the name ?

File: crds/examples/AzureGlobalService.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: networking.aks.io/v1alpha1
+kind: GlobalService # defined by MCS POC
+metadata:
+  name: gwm-mcs-staging # multiClusterService should match the service name / namespace in Child clusters.
```
Comment (akshaysngupta) [inline crds/examples/AzureGlobalService.yaml:4]: call this `service-v2`

File: crds/examples/AzureMultiClusterIngress.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+apiVersion: networking.aks.io/v1alpha1
+kind: AzureMultiClusterIngress
```
Comment (akshaysngupta) [inline crds/examples/AzureMultiClusterIngress.yaml:2]: just call `MultiClusterIngress`

File: pkg/apis/azuremulticlusteringress/v1alpha1/types.go (hunk 1 above)
Comment (erickhernandezascencio) [reply pkg/apis/azuremulticlusteringress/v1alpha1/types.go:23]: sure, what extra properties should we add to spec in nested MultiIngressSpec ?

File: crds/examples/MultiClusterIngress.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+apiVersion: networking.aks.io/v1alpha1
+kind: MultiClusterIngress
+metadata:
+  name: multi-cluster-ingress
+  annotations:
+    kubernetes.io/ingress.class: azure/load-balancer
+spec:
+  rules:
```
Comment (akshaysngupta) [inline crds/examples/MultiClusterIngress.yaml:8]: `template`

**Review by akshaysngupta — CHANGES_REQUESTED** [review]
minor change in example yaml

**Pull request closed without merging**